packages/venia-concept/src/components/Checkout/paymentsForm.js

```diff
@@ -22,228 +21,229 @@ const DEFAULT_FORM_VALUES = {
+                    />
+                </Field>
+            </div>
+            <div className={classes.city}>
+                <Field label="City">
+                    <TextInput
+                        id={classes.city}
+                        field="city"
+                        validate={isRequired}
+                    />
+                </Field>
+            </div>
+            <div className={classes.region_code}>
+                <Field label="State">
+                    <TextInput
+                        id={classes.region_code}
+                        field="region_code"
+                        validate={combine([
+                            isRequired,
+                            [hasLengthExactly, 2],
+                            [validateRegionCode, countries]
+                        ])}
+                    />
+                </Field>
+            </div>
+            <div className={classes.postcode}>
+                <Field label="ZIP">
+                    <TextInput
+                        id={classes.postcode}
+                        field="postcode"
+                        validate={isRequired}
+                    />
+                </Field>
+            </div>
+        </Fragment>
+    );
+};
 
-        let initialFormValues;
-        if (isObjectEmpty(initialValues)) {
-            initialFormValues = DEFAULT_FORM_VALUES;
-        }
-        // We have some initial values, use them.
-        else {
-            if (initialValues.sameAsShippingAddress) {
-                // If the addresses are the same, don't populate any fields
-                // other than the checkbox with an initial value.
-                initialFormValues = {
-                    addresses_same: true
+const FormChildren = props => {
+    const {
+        cancel,
+        classes,
+        countries,
+        isSubmitting,
+        setIsSubmitting,
+        submit,
+        submitting
+    } = props;
+
+    // Currently form state toggles dirty from false to true because of how
+    // informed is implemented. This effectively causes this child components
+    // to re-render multiple times. Keep tabs on the following issue:
+    //   https://github.com/joepuzzo/informed/issues/138
+    // If they resolve it or we move away from informed we can probably get some
+    // extra performance.
+    const formState = useFormState();
+
+    const billingAddressFields = !formState.values.addresses_same ? (
+        <BillingAddressFields classes={classes} countries={countries} />
+    ) : null;
+
+    const handleCancel = useCallback(() => {
+        cancel();
+    }, [cancel]);
+
+    const handleError = useCallback(() => {
+        setIsSubmitting(false);
+    });
+
+    // The success callback. Unfortunately since form state is created first and
+    // then modified when using initialValues any component who uses this
+    // callback will be rendered multiple times on first render. See above
+    // comments for more info.
```

Comment:
Unfortunately, that's the reality of almost any convenient API where we set initial values, validators, etc., on individual components. Those things have to be sent up to the form's state object for actual initialization, meaning it takes a re-render before everything is fully set up.

I think that re-render is acceptable, and it'll be even better if we ever move away from Informed.